frame/curve-amm/src/lib.rs

```diff
@@ -281,6 +279,109 @@ pub mod pallet {
 			PoolCount::<T>::get()
 		}
 
+		fn get_token_index(
+			pool_id: T::PoolId,
+			asset_id: Self::AssetId,
+		) -> Result<Self::PoolTokenIndex, DispatchError> {
+			let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
+			if assets_pair.base == asset_id {
+				Ok(0.into())
+			} else if assets_pair.quote == asset_id {
+				Ok(1.into())
+			} else {
+				Err(Error::<T>::AssetNotFound.into())
```

Comment:
i am bothered with this - we have pair, but have runtime code to handle non pairs..

Reply:
This should probably be removed, see previous comment.